pyrealm/canopy_model/model/community_alt.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import annotations
+
+from pathlib import Path
+
+from numpy.typing import NDArray
+import numpy as np
+
+from pyrealm.canopy_model.model.flora import Flora
+from pyrealm.canopy_model.model.canopy import Canopy
+
+
+class Community:
+    def __init__(
+        self,
+        cell_id: int,
+        cell_area: float,
+        cohort_dbh: NDArray[np.float],
+        cohort_individuals: NDArray[np.int],
+        cohort_pfts: NDArray[np.str],
+        flora: Flora,
+    ):
+        self.cell_id = cell_id
+        self.cell_area = cell_area
+        self.cohort_dbh = cohort_dbh
+        self.cohort_individuals = cohort_individuals
+        self.cohort_pfts = cohort_pfts
+        self.flora = flora
+
+        # Arrays of required PFT traits for cohorts
+        self.traits = CommunityTraits(self.cohort_pfts, self.flora)
+
+        # Calculate geometries from initial dbh
+        self.tmodel_geometries = TModelGeometries(self.cohort_dbh, self.traits)
```

Comment:
We _could_ just have trait arrays and cohort geometries attributes just as part of Community, but it would be a long list and it seems cleaner and more easily tested if we encapsulate them.

Reply:
Thanks David, it's really good to see an illustration of what you meant. I'm a bit concerned that when adding and removing cohorts we'd have to be very careful to ensure that we removed the relevant entries from every set of traits, ie all the t model geometries, all the community traits, etc., since if we missed one, we'd end up with the data misaligned, which might make it a bit fragile. If we later added new attributes to a cohort, then we'd also need to be sure we were adding them into whatever function we use to remove/add them. The best case would be the arrays being different lengths and ending up with a runtime failure, but I'd also be worried about the program silently just attaching the wrong geometry to the wrong cohort etc (for example if we removed one cohort from one data set and a different one from another).

Reply:
That is a clear advantage of the OO `Cohort` model - and also of using some kind of record array. Having said that, we could also write a fairly defensive `remove_cohort(index)` method to these classes.